nats/replica_client.go

```diff
@@ -0,0 +1,402 @@
+package nats
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/benbjohnson/litestream"
+	"github.com/benbjohnson/litestream/internal"
+	natsClient "github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+	"github.com/valyala/bytebufferpool"
+)
+
+const (
+	ReplicaClientType     = "nats"
+	DefaultBucketName     = "litestream"
+	DefaultBucketReplicas = 3
+	DefaultBucketMaxBytes = 0               // unlimited
+	DefaultBucketTTL      = 0 * time.Second // unlimited
+
+	generationsPrefix    = "generations/"
+	generationsPrefixLen = len(generationsPrefix)
+)
+
+type ReplicaClient struct {
+	nc          *natsClient.Conn
+	objectStore jetstream.ObjectStore
+
+	// NATS connection info
+	BucketName     string
+	BucketReplicas int
+	BucketMaxBytes int64
+	BucketTTL      time.Duration
+}
+
+func NewReplicaClient(nc *natsClient.Conn) *ReplicaClient {
+	return &ReplicaClient{
+		nc:             nc,
+		BucketName:     DefaultBucketName,
+		BucketReplicas: DefaultBucketReplicas,
+		BucketMaxBytes: DefaultBucketMaxBytes,
+		BucketTTL:      DefaultBucketTTL,
+	}
+}
+
+// Returns the type of client.
```

Comment:
Go method and type comments should start with the name of the method or type:

```
// Type returns the type of client.
```

instead of

```
// Returns the type of client. 
```